crates/biome_js_analyze/src/lint/nursery/no_floating_promises.rs

```diff
@@ -95,22 +106,29 @@ impl Rule for NoFloatingPromises {
         let node = ctx.query();
         let model = ctx.model();
         let expression = node.expression().ok()?;
-        if let AnyJsExpression::JsCallExpression(js_call_expression) = expression {
-            let Ok(any_js_expression) = js_call_expression.callee() else {
-                return None;
-            };
+        match expression {
+            AnyJsExpression::JsCallExpression(js_call_expression) => {
+                let any_js_expression = js_call_expression.callee().ok()?;
 
-            if !is_callee_a_promise(&any_js_expression, model)? {
-                return None;
-            }
+                if !is_callee_a_promise(&any_js_expression, model)? {
+                    return None;
+                }
+
+                if is_handled_promise(&js_call_expression).unwrap_or(false) {
```

Comment:
```suggestion
                if is_handled_promise(&js_call_expression)? {
```

Reply:
I actually want to return only when the value is true and continue when the value is false or None so using `?` does not work here